codegen/editable_token_from_data.php

```diff
@@ -15,419 +15,213 @@ function editable_token_from_data(
+    '[' => HHAST\LeftBracketToken::class,
+    ']' => HHAST\RightBracketToken::class,
+    '(' => HHAST\LeftParenToken::class,
+    ')' => HHAST\RightParenToken::class,
+    '{' => HHAST\LeftBraceToken::class,
+    '}' => HHAST\RightBraceToken::class,
+    '.' => HHAST\DotToken::class,
+    '->' => HHAST\MinusGreaterThanToken::class,
+    '++' => HHAST\PlusPlusToken::class,
+    '--' => HHAST\MinusMinusToken::class,
+    '**' => HHAST\StarStarToken::class,
+    '*' => HHAST\StarToken::class,
+    '+' => HHAST\PlusToken::class,
+    '-' => HHAST\MinusToken::class,
+    '~' => HHAST\TildeToken::class,
+    '!' => HHAST\ExclamationToken::class,
+    '$' => HHAST\DollarToken::class,
+    '/' => HHAST\SlashToken::class,
+    '%' => HHAST\PercentToken::class,
+    '<>' => HHAST\LessThanGreaterThanToken::class,
+    '<=>' => HHAST\LessThanEqualGreaterThanToken::class,
+    '<<' => HHAST\LessThanLessThanToken::class,
+    '>>' => HHAST\GreaterThanGreaterThanToken::class,
+    '<' => HHAST\LessThanToken::class,
+    '>' => HHAST\GreaterThanToken::class,
+    '<=' => HHAST\LessThanEqualToken::class,
+    '>=' => HHAST\GreaterThanEqualToken::class,
+    '==' => HHAST\EqualEqualToken::class,
+    '===' => HHAST\EqualEqualEqualToken::class,
+    '!=' => HHAST\ExclamationEqualToken::class,
+    '!==' => HHAST\ExclamationEqualEqualToken::class,
+    '^' => HHAST\CaratToken::class,
+    '|' => HHAST\BarToken::class,
+    '&' => HHAST\AmpersandToken::class,
+    '&&' => HHAST\AmpersandAmpersandToken::class,
+    '||' => HHAST\BarBarToken::class,
+    '?' => HHAST\QuestionToken::class,
+    '??' => HHAST\QuestionQuestionToken::class,
+    ':' => HHAST\ColonToken::class,
+    ';' => HHAST\SemicolonToken::class,
+    '=' => HHAST\EqualToken::class,
+    '**=' => HHAST\StarStarEqualToken::class,
+    '*=' => HHAST\StarEqualToken::class,
+    '/=' => HHAST\SlashEqualToken::class,
+    '%=' => HHAST\PercentEqualToken::class,
+    '+=' => HHAST\PlusEqualToken::class,
+    '-=' => HHAST\MinusEqualToken::class,
+    '.=' => HHAST\DotEqualToken::class,
+    '<<=' => HHAST\LessThanLessThanEqualToken::class,
+    '>>=' => HHAST\GreaterThanGreaterThanEqualToken::class,
+    '&=' => HHAST\AmpersandEqualToken::class,
+    '^=' => HHAST\CaratEqualToken::class,
+    '|=' => HHAST\BarEqualToken::class,
+    ',' => HHAST\CommaToken::class,
+    '@' => HHAST\AtToken::class,
+    '::' => HHAST\ColonColonToken::class,
+    '=>' => HHAST\EqualGreaterThanToken::class,
+    '==>' => HHAST\EqualEqualGreaterThanToken::class,
+    '?->' => HHAST\QuestionMinusGreaterThanToken::class,
+    '...' => HHAST\DotDotDotToken::class,
+    '$$' => HHAST\DollarDollarToken::class,
+    '|>' => HHAST\BarGreaterThanToken::class,
+    '/>' => HHAST\SlashGreaterThanToken::class,
+    '</' => HHAST\LessThanSlashToken::class,
+    '<?' => HHAST\LessThanQuestionToken::class,
+    '?>' => HHAST\QuestionGreaterThanToken::class,
+  ];
+  $cls = idx($class_map, $token_kind);
```

Comment:
`$class_map[$token_kind] ?? null`

Reply:
yields
```
codegen/editable_token_from_data.php:220:16,19: Was expecting a class, but got an untyped value (Typing[4026])
codegen/editable_token_from_data.php:224:16,19: Was expecting a class, but got an untyped value (Typing[4026])
```
(on `3.22.1-dev`)
:/

Reply:
:/ is this the full line that you have?

```
$cls = $class_map[$token_kind] ?? null;
```

Reply:
yessir, generated:
```
  $cls = $class_map[$token_kind] ?? null;
  if ($cls !== null) {
    return new $cls($leading, $trailing);
  }
  $cls = $class_map_with_text[$token_kind] ?? null;
  if ($cls !== null) {
    return new $cls($leading, $trailing, $token_text);
  }
```
`hh_client` doesn't like either of the `return` lines at `$cls`

Reply:
okay - just a small nit, let's not do this change for now then.

Reply:
Actually, how is this valid? There's no common superclass defining a consistent constructor for these? That's likely why this is complaining